velox/connectors/hive/HiveConnectorUtil.cpp

```diff
@@ -383,18 +383,16 @@ std::shared_ptr<common::ScanSpec> makeScanSpec(
     }
   }
 
-  int numChildren = 0;
   // Process columns that will be projected out.
   for (int i = 0; i < rowType->size(); ++i) {
     auto& name = rowType->nameOf(i);
     auto& type = rowType->childAt(i);
     if (isRowIndexColumn(name, rowIndexColumn)) {
       VELOX_CHECK(type->isBigint());
-      continue;
     }
     auto it = outputSubfields.find(name);
     if (it == outputSubfields.end()) {
-      auto* fieldSpec = spec->addFieldRecursively(name, *type, numChildren++);
+      auto* fieldSpec = spec->addFieldRecursively(name, *type, i);
```

Comment:
We can do `fieldSpec->setExplicitRowNumber` here and L410 so we don't need to do it in `SplitReader::adaptColumns`.  In general we don't want to mutate `ScanSpec` per split too much.

Reply:
I see, updated, thanks.